example/src/main/java/io/netty/example/tunechoserver/TunEchoServer.java

```diff
@@ -0,0 +1,452 @@
+ *     |00000040| 10 11 12 13 14 15 16 17 18 19 1a 1b 1c 1d 1e 1f |................|
+ *     |00000050| 20 21 22 23 24 25 26 27 28 29 2a 2b 2c 2d 2e 2f | !"#$%&'()*+,-./|
+ *     |00000060| 30 31 32 33 34 35 36 37                         |01234567        |
+ *     +--------+-------------------------------------------------+----------------+
+ *
+ *     Feb 01, 2017 7:26:16 PM io.netty.channel.epoll.TunTapPacketLogger logPacket
+ *     INFO: TUN/TAP PACKET SENT (protocol IPv6, len 104):
+ *       IPv6: src fd00:0:1:0:0:0:0:2, dest fd00:0:1:0:0:0:0:1, protocol ICMPv6
+ *              +-------------------------------------------------+
+ *              |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+ *     +--------+-------------------------------------------------+----------------+
+ *     |00000000| 60 03 62 2f 00 40 3a 40 fd 00 00 00 00 01 00 00 |`.b/.@:@........|
+ *     |00000010| 00 00 00 00 00 00 00 02 fd 00 00 00 00 01 00 00 |................|
+ *     |00000020| 00 00 00 00 00 00 00 01 81 00 5a 5c 19 8a 00 01 |..........Z\....|
+ *     |00000030| d8 a6 92 58 00 00 00 00 da c4 0c 00 00 00 00 00 |...X............|
+ *     |00000040| 10 11 12 13 14 15 16 17 18 19 1a 1b 1c 1d 1e 1f |................|
+ *     |00000050| 20 21 22 23 24 25 26 27 28 29 2a 2b 2c 2d 2e 2f | !"#$%&'()*+,-./|
+ *     |00000060| 30 31 32 33 34 35 36 37                         |01234567        |
+ *     +--------+-------------------------------------------------+----------------+
+ * </pre>
+ *
+ * 5) To test UDP, use the netcat (nc) tool to send and receive UDP packets from the server.
+ *
+ * <pre>
+ *     echo "HELLO via UDP!" | nc -u 172.20.0.2 7
+ *
+ *     echo "HELLO via UDP!" | nc -u -6 fd00:0:1::2 7
+ * </pre>
+ *
+ *
+ * <h3>Configurable Properties</h3>
+ *
+ * TunEchoServer supports the following system properties for configuring its behavior:
+ *
+ * tun-device=<device-name>
+ *
+ * Open the specified linux TUN device. Defaults to 'tun-echo-server'.
+ *
+ * thread-pool-size=<int>
+ *
+ * Use a ThreadPoolExecutor with the specified number of threads to respond to incoming packets.
+ * A value of 0 causes the responses to be sent on the Netty channel thread. Defaults to 0.
+ *
+ * use-unpooled=<true|false>
+ *
+ * Enable/disable use of Netty UnpooledByteBufAllocator for allocating buffers. Defaults to false.
+ *
+ * read-buf-size=<int>
+ *
+ * Set the TunTapChannel read buffer size.  Defaults to 2048.
+ *
+ * force-gc=<true|false>
+ *
+ * Enable/disable garbage collection and finalizers after each packet is processed. Defaults to false.
+ *
+ * debug=<true|false>
+ *
+ * Enable/disable verbose logging of server activity. Disabling debug logging is using running
+ * performance tests.  Defaults to true.
+ *
+ */
+
+public class TunEchoServer extends SimpleChannelInboundHandler<TunTapPacket> {
+
+    private static AbstractByteBufAllocator _bufAllocator = PooledByteBufAllocator.DEFAULT;
+    private static ThreadPoolExecutor _threadPool;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this field "_threadPool" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS3008)